src/components/GroupsTable/GroupsTable.js

```diff
@@ -0,0 +1,237 @@
+import upperCase from 'lodash/upperCase';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { TABLE_DEFAULT_PAGINATION } from '../../constants';
+import { fetchGroups } from '../../store/inventory-actions';
+import { generateLoadingRows } from '../InventoryTable/helpers';
+import NoEntitiesFound from '../InventoryTable/NoEntitiesFound';
+
+const GROUPS_TABLE_INITIAL_STATE = {
+    perPage: TABLE_DEFAULT_PAGINATION,
+    page: 1
+};
+
+const GROUPS_TABLE_COLUMNS = [
+    {
+        title: 'Name',
+        transforms: [sortable, cellWidth(40)]
+    },
+    {
+        title: 'Total systems',
+        transforms: [sortable, cellWidth(20)]
+    },
+    {
+        title: 'Last modified',
+        transforms: [sortable, cellWidth(20)]
+    }
+];
+
+const GROUPS_TABLE_COLUMNS_TO_URL = {
+    0: 'name',
+    1: 'host_ids',
+    2: 'updated_at'
+};
+
+const REQUEST_DEBOUNCE_TIMEOUT = 500;
+
+const GroupsTable = () => {
+    const dispatch = useDispatch();
+    const { rejected, uninitialized, loading, data } = useSelector(
+        (state) => state.groups
+    );
+    const [filters, setFilters] = useState(GROUPS_TABLE_INITIAL_STATE);
+    const [rows, setRows] = useState([]);
+    const groups = useMemo(() => data?.results || [], [data]);
+
+    const fetchData = useCallback(
+        debounce((filters) => {
+            const { perPage, page, sortIndex, sortDirection, ...search } = filters;
+
+            if (sortIndex !== undefined && sortDirection !== undefined) {
+                const order_by = GROUPS_TABLE_COLUMNS_TO_URL[sortIndex];
+                const order_how = upperCase(sortDirection);
+                return dispatch(
+                    fetchGroups({ ...search, order_by, order_how }, { page, perPage })
+                );
+            } else {
+                return dispatch(fetchGroups(search, { page, perPage }));
+            }
+        }, REQUEST_DEBOUNCE_TIMEOUT), // wait the timeout before making the final fetch
+        []
+    );
+
+    useEffect(() => {
+        fetchData(filters);
+    }, [filters]);
+
+    useEffect(() => {
+        // update visible rows once new data obtained
+        const newRows = groups.map((group, index) => ({
+            cells: [
+                <span key={index}>
+                    <Link to={`groups/${group.id}`}>{group.name || group.id}</Link>
+                </span>,
+                <span key={index}>{(group.host_ids || []).length.toString()}</span>,
+                <span key={index}>{<DateFormat date={group.updated_at} />}</span>
+            ]
+        }));
+        setRows(newRows);
+    }, [groups]);
+
+    // TODO: convert initial URL params to filters
+
+    const onSort = (event, index, direction) => {
+        setFilters({ ...filters, sortIndex: index, sortDirection: direction });
+    };
+
+    const filterConfigItems = [
```

Comment:
I am wondering if there is a way to create a more dynamic function to create filters and filter configs... This may be a starting point to improve the filtering on the Inventory table as well.

Now, the SearchInput component is re-created every time on re-render.

Reply:
good point! I think I'll wrap it with useMemo since it looks like a good example where it can become useful